### ISSUE TITLE
Error require --harmony in command

### ISSUE THREAD

**Issue by nguyenchilong** [nodejs/help#39]
throw new Error ('AppJS requires Node is run with the --harmony command line flag');  ^
Error: AppJS requires Node is run with the --harmony command line flag
    at Object.<anonymous> (/var/www/tutorial/app/node_modules/appjs/lib/index.js:2:9)
    at Module._compile (module.js:435:26)
    at Object.Module._extensions..js (module.js:442:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:311:12)
    at Module.require (module.js:366:17)
    at require (module.js:385:17)
    at Object.<anonymous> (/var/www/tutorial/app/app.js:3:11)
    at Module._compile (module.js:435:26)
    at Object.Module._extensions..js (module.js:442:10)

Please help me fix it, so I just start with node js and AppJS for new project alternative for java.

**Comment by bnoordhuis** [comment]
Have you tried `node --harmony app/app.js`?

**Comment by Trott** [comment]
Ref: https://github.com/nodejs/node/issues/4070

**Comment by nguyenchilong** [comment]
bnoordhuis: "Have you tried node --harmony app/app.js?"
I tried with it but the same error.

**Comment by Trott** [comment]
You might have better luck with `--harmony-proxies` in addition to `--harmony`.

That said, instead of AppJS, you might want to look at a more current project that does the same thing like [Electron](http://electron.atom.io/)

As noted by @ChALkeR in #4070: 

---

...AppJS itself looks deprecated and unsupported, from [their GitHub page](https://github.com/appjs/appjs):
> (Deprecated!) SDK on top of nodejs to build desktop apps using HTML5/CSS/JS http://appjs.com

And the last commit was about a year ago with a typo fix in the Readme, the previous to that commit was more than two years ago.

---

In short, AppJS doesn't look like a good choice for a new project and should be expected to have all sorts of problems. Even appjs.com says it's a work in progress and has a 2012 copyright.

**Comment by ChALkeR** [comment]
Copy-pasting my answer here from https://github.com/nodejs/node/issues/4070#issuecomment-160568971:

>AppJS checks for `Proxy` and `WeakMap`. `WeakMap` is present without `--harmony`, `Proxy` requires `--harmony-proxies` and is not enabled by just `--harmony`.
Ref: https://github.com/appjs/appjs/blob/master/lib/index.js#L1.
Also check https://nodejs.org/en/docs/es6/.
>
>Note this part about «in progress» features (as `--harmony-proxies`) :
>> this is highly discouraged unless for testing purposes.
>
>--
>
>Also note that AppJS itself looks deprecated and unsupported, from [their GitHub page](https://github.com/appjs/appjs):
>> (Deprecated!) SDK on top of nodejs to build desktop apps using HTML5/CSS/JS http://appjs.com
>
>And the last commit was about a year ago with a typo fix in the Readme, the previous to that commit was more than two years ago.

**Comment by eladkarako** [comment]
<h4><em>Sharing the pain...</em></h4>
@nguyenchilong In most of the cases the generic app will try to use newer features such as <code>Proxy</code> or <code>WeakMap</code>, 

<strong>The thing is, harmony-switches are no longer being used</strong> in <a href="https://nodejs.org/download/nightly/v12.0.0-nightly201903091acf3b155f/"><strong>newer</strong> versions of node</a> (at least you can't find it under the `--help` entries), so even if you'll add `--harmony`, `--harmony_proxies`, `--harmony_collections` and `--harmony_scoping` it will do nothing..

A malformed check is to see if <code>Proxy</code> is a type-of <code>object</code> (used to be, is <code>function</code> now..).
 
One such application is <a href="https://github.com/appjs/appjs">appjs</a> (sadly it is not maintained anymore...)
<img src="https://i.imgur.com/FJxmn2Wl.jpg" />
<img src="https://i.imgur.com/nMl2x66.jpg" />
<img src="https://i.imgur.com/gBeDnJc.jpg" />

That "check part" is seems to be very common (probably <em>copy-pasted</em> from an old project or "bootstrapped"-in)

As far as I can see, you can either manually locate, and <strong>comment-out</strong> the checking-sections and re-run your app. It might work.
For example:
<img src="https://i.imgur.com/WkObmHS.jpg" />

or, if possible, open a bug and let the author handle the compatibility issue.
